File: Easy/0001_two-sum/tow-sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for i, n := range nums {
		comp := target - n
		if idx, ok := seen[comp]; ok {
			return []int{idx, i}
		}
		seen[n] = i
	}
	return nil
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9;
	result := twoSum(nums, target)
	fmt.Println(result)
}

File: Easy/0001_two-sum/tow-sum.py
#!/usr/bin/env python3

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen: dict[int, int] = {}
        for i, n in enumerate(nums):
            comp = target - n
            if comp in seen:
                return [seen[comp], i]
            seen[n] = i
        return []
